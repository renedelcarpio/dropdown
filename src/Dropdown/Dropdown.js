import Loading from '../Loading/Loading';
import axios from 'axios';
import { useState } from 'react';

const baseURL = 'https://jsonplaceholder.typicode.com/posts';

const Dropdown = () => {
	const [posts, setPosts] = useState(null);
	const [loading, setLoading] = useState(false);

	const getData = async () => {
		const { data } = await axios.get(baseURL);
		setPosts(data);
		if (!data) {
			setLoading(true);
		} else {
			setLoading(false);
		}
	};

	return (
		<div>
			<button onClick={getData}>get data</button>
			{loading && <Loading />}
			{posts &&
				posts.map(({ title, body, id }) => {
					return (
						<div key={id}>
							<h3>{title}</h3>
							<p>{body}</p>
							<br />
						</div>
					);
				})}
		</div>
	);
};

export default Dropdown;
