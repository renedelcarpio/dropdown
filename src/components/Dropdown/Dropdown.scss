.dropdown__button {
  background-color: transparent;
  margin: 2rem;
  border: none;
}

.dropdown__data {
  display: flex;
  margin-bottom: 1rem;
}

.dropdown__data--item {
  margin-right: 1rem;
  padding: 1rem;
}
