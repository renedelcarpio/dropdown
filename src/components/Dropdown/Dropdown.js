import './Dropdown.scss';

import DropdownMenu from '../DropdownMenu/DropdownMenu';
import InfiniteScroll from 'react-infinite-scroll-component';
import axios from 'axios';
import { useState } from 'react';

const Dropdown = () => {
	const baseURL =
		'https://back.implementaconbubo.com/v1/sales/customer/?simple=true';

	const APIKEY = process.env.REACT_APP_API_KEY;

	const AuthAxios = axios.create({
		url: baseURL,
		headers: {
			Authorization: `ApiKey ${APIKEY}`,
			DataOperations: {
				filter: {
					filters: [
						{
							field: 'reference_name' || selectedOption,
							operator: 'contains',
							ignoreCase: true,
							value: '',
						},
					],
					logic: 'or',
				},
				skip: 0,
				take: 5,
			},
		},
	});

	const [results, setResults] = useState([]);
	const [currentData, setCurrentData] = useState(0);
	const [selectedOption, setSelectedOption] = useState(null);

	console.log(selectedOption);

	const getData = async () => {
		const { results } = await AuthAxios.get(baseURL);
		setResults(results);
	};

	const filteredData = () => {
		return results.slice(currentData, currentData + 20);
	};

	const loadMoreData = () => {
		setCurrentData(currentData + 10);
	};

	return (
		<div>
			<button className='dropdown__button' onClick={getData}>
				<DropdownMenu
					setSelectedOption={setSelectedOption}
					value={selectedOption}
				/>
			</button>
			{results &&
				filteredData().map(({ id, reference_name, nit, name }) => {
					return (
						<InfiniteScroll
							key={id}
							dataLength={results.length}
							hasMore={true}
							next={loadMoreData}
						>
							<div className='dropdown__data'>
								<p className='dropdown__data--item'>{id}</p>
								<p className='dropdown__data--item'>{reference_name}</p>
								<p className='dropdown__data--item'>{name}</p>
								<p className='dropdown__data--item'>{nit}</p>
							</div>
						</InfiniteScroll>
					);
				})}
		</div>
	);
};

export default Dropdown;
