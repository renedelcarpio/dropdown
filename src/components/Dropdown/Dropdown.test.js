import Dropdown from './Dropdown';
import { mount } from 'enzyme';

describe('Tests in Dropdown component', () => {
	const wrapper = mount(<Dropdown />);

	test('should match with snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should have the DropdownMenu component', () => {
		expect(wrapper.find('DropdownMenu').exists()).toBe(true);
	});
});
