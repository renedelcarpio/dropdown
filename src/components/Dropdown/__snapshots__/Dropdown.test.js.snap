// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Dropdown component should match with snapshot 1`] = `
<div>
  <button
    className="dropdown__button"
    onClick={[Function]}
  >
    <div
      className="dropdown__menu"
      onClick={[Function]}
    >
      Dropdown Menu
    </div>
  </button>
</div>
`;
